### ISSUE TITLE
Multiplayer broken

### ISSUE THREAD

**Issue by Dennispandea** [Filostorm/Multiblock-Madness-2#7]
Hi, I decided to create a server using the modpack files for me and a friend to play on. This didn't work. Here is the info on what happened.

https://pastebin.com/Fb2xW1jz

The same thing happened when trying to join with the essential mod

https://pastebin.com/ATQtjkx5

And when trying to join through a LAN session, you just get the "disconnected" error

**Comment by Filostorm** [comment]
I can't see those pastebins for some reason

**Comment by Frefreak** [comment]
We encountered this too when trying to setup a dedicated server:
server log:
```
[23:44:56] [Server thread/INFO] [minecraft/ServerGamePacketListenerImpl]: ... lost connection: Internal Exception: io.netty.handler.codec.EncoderException: net.minecraft.ResourceLocationException: Non [a-z0-9_.-] character in na
mespace of location: mekanism:crushing$mekanism:crushing/dewax/copper_blockUBminecraft:crafting_shapedthermal:rich_slag_bricks�2�2�2�2�2mekanism:crushing?......
```
client log:
https://mclo.gs/6oSsz6l

**Comment by Frefreak** [comment]
The encoding error seems to come from `kubejs/server_scripts/multiblocked/air_collector.js`. Finally after I remove this single file in server (or commented out all the content in this file) I managed to make the players connect and starts playing (with some kubejs errors and warnings).

However I don't know how robust this will be and I will let the server run for a few days to see how it goes. The air collector will definitely be broken after this change so could you please take a look at this if you find time? MBM2 looks like an interesting pack and it will be fun if the multi-player setup could be easier in the future.

BTW this is the one kubejs error it shows if I run `kubejs errors` in multiplayer so it might has potential issues too.
```
[20:39:01] [ERR  ] Error occurred while handling event 'recipes': TypeError: Cannot read property "1" from undefined (server_scripts:ore_processing.js#355)
```

Also those are the warnings:
```
[20:38:52] [WARN ] Failed to register handler for recipe type ars_nouveau:glyph_recipe as it doesn't exist!
[20:38:59] [WARN ] Failed to parse recipe 'biggerreactors:compat/mekanism/rotary/superheated_sodium[mekanism:rotary]'! Falling back to vanilla: com.google.gson.JsonSyntaxException: Invalid fluid type 'biggerreactors:superheated_sodium'
[20:38:59] [WARN ] Failed to parse recipe biggerreactors:compat/mekanism/rotary/superheated_sodium[mekanism:rotary]: com.google.gson.JsonSyntaxException: Invalid fluid type 'biggerreactors:superheated_sodium'
[20:38:59] [WARN ] Failed to parse recipe 'beyond_earth:smeltery/melting/metal/calorite/ore[tconstruct:ore_melting]'! Falling back to vanilla: com.google.gson.JsonSyntaxException: Missing rate, expected to find a string
[20:38:59] [WARN ] Failed to parse recipe beyond_earth:smeltery/melting/metal/calorite/ore[tconstruct:ore_melting]: com.google.gson.JsonSyntaxException: Missing rate, expected to find a string
[20:38:59] [WARN ] Failed to parse recipe 'beyond_earth:smeltery/melting/metal/calorite/raw[tconstruct:ore_melting]'! Falling back to vanilla: com.google.gson.JsonSyntaxException: Missing rate, expected to find a string
[20:38:59] [WARN ] Failed to parse recipe beyond_earth:smeltery/melting/metal/calorite/raw[tconstruct:ore_melting]: com.google.gson.JsonSyntaxException: Missing rate, expected to find a string
[20:38:59] [WARN ] Failed to parse recipe 'thermal:machines/centrifuge/centrifuge_oil_sand[thermal:centrifuge]': {"fluid":"pneumaticcraft:crude_oil","amount":100} is not a valid fluid!
[20:38:59] [WARN ] Failed to parse recipe 'beyond_earth:smeltery/melting/metal/desh/raw_block[tconstruct:ore_melting]'! Falling back to vanilla: com.google.gson.JsonSyntaxException: Missing rate, expected to find a string
[20:38:59] [WARN ] Failed to parse recipe beyond_earth:smeltery/melting/metal/desh/raw_block[tconstruct:ore_melting]: com.google.gson.JsonSyntaxException: Missing rate, expected to find a string
[20:38:59] [WARN ] Failed to parse recipe 'createaddition:compat/tconstruct/tinkers_bronze[create:mixing]': {"item":"tconstruct:tinkers_bronze_ingot","count":3} is not a valid result!
[20:38:59] [WARN ] Failed to parse recipe 'beyond_earth:smeltery/melting/metal/calorite/raw_block[tconstruct:ore_melting]'! Falling back to vanilla: com.google.gson.JsonSyntaxException: Missing rate, expected to find a string
[20:38:59] [WARN ] Failed to parse recipe beyond_earth:smeltery/melting/metal/calorite/raw_block[tconstruct:ore_melting]: com.google.gson.JsonSyntaxException: Missing rate, expected to find a string
[20:38:59] [WARN ] Failed to parse recipe 'beyond_earth:smeltery/melting/metal/ostrum/raw[tconstruct:ore_melting]'! Falling back to vanilla: com.google.gson.JsonSyntaxException: Missing rate, expected to find a string
[20:38:59] [WARN ] Failed to parse recipe beyond_earth:smeltery/melting/metal/ostrum/raw[tconstruct:ore_melting]: com.google.gson.JsonSyntaxException: Missing rate, expected to find a string
[20:38:59] [WARN ] Failed to parse recipe 'beyond_earth:smeltery/melting/metal/desh/ore[tconstruct:ore_melting]'! Falling back to vanilla: com.google.gson.JsonSyntaxException: Missing rate, expected to find a string
[20:38:59] [WARN ] Failed to parse recipe beyond_earth:smeltery/melting/metal/desh/ore[tconstruct:ore_melting]: com.google.gson.JsonSyntaxException: Missing rate, expected to find a string
[20:38:59] [WARN ] Failed to parse recipe 'immersiveengineering:blueprint/component_electronic[unknown:type]'! Falling back to vanilla: com.google.gson.JsonSyntaxException: Missing type, expected to find a string
[20:38:59] [WARN ] Failed to parse recipe immersiveengineering:blueprint/component_electronic[unknown:type]: com.google.gson.JsonSyntaxException: Missing type, expected to find a string
[20:38:59] [WARN ] Failed to parse recipe 'beyond_earth:smeltery/melting/metal/desh/raw[tconstruct:ore_melting]'! Falling back to vanilla: com.google.gson.JsonSyntaxException: Missing rate, expected to find a string
[20:38:59] [WARN ] Failed to parse recipe beyond_earth:smeltery/melting/metal/desh/raw[tconstruct:ore_melting]: com.google.gson.JsonSyntaxException: Missing rate, expected to find a string
[20:38:59] [WARN ] Failed to parse recipe 'immersiveengineering:blueprint/component_electronic_adv[unknown:type]'! Falling back to vanilla: com.google.gson.JsonSyntaxException: Missing type, expected to find a string
[20:38:59] [WARN ] Failed to parse recipe immersiveengineering:blueprint/component_electronic_adv[unknown:type]: com.google.gson.JsonSyntaxException: Missing type, expected to find a string
[20:39:00] [WARN ] Failed to parse recipe 'immersiveengineering:blueprint/electron_tube[unknown:type]'! Falling back to vanilla: com.google.gson.JsonSyntaxException: Missing type, expected to find a string
[20:39:00] [WARN ] Failed to parse recipe immersiveengineering:blueprint/electron_tube[unknown:type]: com.google.gson.JsonSyntaxException: Missing type, expected to find a string
[20:39:00] [WARN ] Failed to parse recipe 'pneumaticcraft:pneumaticcraft_fuels/cofh_biofuel[pneumaticcraft:fuel_quality]'! Falling back to vanilla: com.google.gson.JsonSyntaxException: Unknown fluid 'thermal:refined_biofuel'
[20:39:00] [WARN ] Failed to parse recipe pneumaticcraft:pneumaticcraft_fuels/cofh_biofuel[pneumaticcraft:fuel_quality]: com.google.gson.JsonSyntaxException: Unknown fluid 'thermal:refined_biofuel'
[20:39:00] [WARN ] Failed to parse recipe 'beyond_earth:smeltery/melting/metal/ostrum/raw_block[tconstruct:ore_melting]'! Falling back to vanilla: com.google.gson.JsonSyntaxException: Missing rate, expected to find a string
[20:39:00] [WARN ] Failed to parse recipe beyond_earth:smeltery/melting/metal/ostrum/raw_block[tconstruct:ore_melting]: com.google.gson.JsonSyntaxException: Missing rate, expected to find a string
[20:39:00] [WARN ] Failed to parse recipe 'beyond_earth:smeltery/melting/metal/ostrum/ore[tconstruct:ore_melting]'! Falling back to vanilla: com.google.gson.JsonSyntaxException: Missing rate, expected to find a string
[20:39:00] [WARN ] Failed to parse recipe beyond_earth:smeltery/melting/metal/ostrum/ore[tconstruct:ore_melting]: com.google.gson.JsonSyntaxException: Missing rate, expected to find a string
```

**Comment by Filostorm** [comment]
Try the newest update, it has the latest version of multiblocked, which should have fixed this issue

**Comment by Frefreak** [comment]
Thanks, the newest update works :smile:

**Comment by Filostorm** [comment]
Awesome, so just to help out our server staff, you are able to play without doing anything drastic? 

Would you mind listing your steps? 

I need to start being able to build a server pack

**Comment by Frefreak** [comment]
Sure, for the newest version (0.2.3) this is what I did (a more detailed version than my previous comment):

- download the 0.2.3 zip and extract, copy the content of `overrides` to a new folder, which will be the root of my server pack
- download forge installer from [forge](https://files.minecraftforge.net/net/minecraftforge/forge/index_1.18.2.html) and install the server files. The client uses 40.2.9 so I used the same version.
- copy all mods from the pack installed by my launcher, then for 0.2.3 I need to delete this mod file: `LegendaryTooltips-1.18.2-1.3.1.jar` for the server to run
- this is the hardest step for me. The mod "fluidizedtanks-0.1.jar" tries to load client side class even in server side, so it will crash. Luckily it is open sourced and the fix is relatively straight-forward.
The relevant PR is [here](https://github.com/Rongmario/Fluidized-Tanks/pull/3), or if you trust me you can try my build before author accepts the PR or make their own official fix.
[jars.zip](https://github.com/Filostorm/Multiblock-Madness-2/files/11958707/jars.zip)

After those steps the server should be able to run.

**Comment by Filostorm** [comment]
Everything should be working properly with the upcoming version, and the plan is to release an update along with server patch today or tomorrow
Fixes #8 as well
